nemo/collections/nlp/modules/common/common_utils.py

```diff
@@ -59,10 +59,11 @@ def get_pretrained_lm_model(
     else:
         if pretrained_model_name in get_megatron_lm_models_list():
             model, default_checkpoint_file = get_megatron_lm_model(
-                config_file=config_file, pretrained_model_name=pretrained_model_name
+                config_dict=config_dict,
+                config_file=config_file,
```

Comment:
why do we need config_dict to Megatron? Megatron doesn't support it

Reply:
I think we should be able to configure Megatron from yaml. I hate to force a user to pass in a .json file when we already have a yaml.

Reply:
do we have a working yaml megatron config? are you proposing to change all default model's configs to support this?

Reply:
No, I was just trying to unify interface for the two models. If I was to run all the tests written for HF with Megatron, I would like it to be able to take same arguments.

Reply:
what I'm unclear about is where does this separate yaml is coming from? for .json file - it is released on NGC along with the pretrained megatron checkpoint, default yaml model files won't contain megatron configs, so where the user is going to get it?

Reply:
@ekmb I have some ideas on how we can do this really nicely for NeMo users. But that is beyond the scope of this PR. For this PR, if we have config_dict option, then we can put megatron config into the yaml like we did for bert pretraining: https://github.com/NVIDIA/NeMo/blob/candidate/examples/nlp/language_modeling/conf/bert_pretraining_from_text_config.yaml

Reply:
Note that, we don't have to put anything into the yaml, and we can specify the .json file as the config_file has precedence. By having config_dict we have more flexibility and I don't think it's harming anything.

Reply:
@ericharper we can't add megatron config to every model configs - this is just impractical. 
I'm fine with leaving this as-is, it just might confuse users imo